demisto_sdk/dev_tools/spell_checker.py

```diff
@@ -0,0 +1,148 @@
+import ssl
+import yaml
+import nltk
+from typing import Set
+from spellchecker import SpellChecker
+
+from nltk.corpus import words
+from argparse import ArgumentDefaultsHelpFormatter
+from demisto_sdk.common.constants import KNOWN_WORDS
+from demisto_sdk.common.tools import print_error, print_color, LOG_COLORS
+
+
+# These are keys in a Demisto yml file which indicate that their values are visible to the user and
+# thus their spelling should be checked.
+USER_VISIBLE_LINE_KEYS = [
+    "description",
+    "display",
+    "name",
+    "comment"
+]
+
+
+class SpellCheck:
+    """Perform a spell check on the given .yml or .md file.
+        Attributes:
+            checked_file_path (str): The path to the current file being checked.
+            known_words (str): The path to a file containing known words.
+            spellchecker (SpellChecker): The spell-checking object.
+            unknown_words (set): A set of unknown words found in the given file.
+    """
+
+    def __init__(self, checked_file_path: str, known_words: str):
+        self.checked_file_path = checked_file_path
+        self.spellchecker = SpellChecker()
+        self.unknown_words = set([])  # type:Set
+        self.known_words = known_words
+
+    def run_spell_check(self):
+        """Runs spell-check on the given file.
+
+        Returns:
+            bool. True if no problematic words found, False otherwise.
+        """
+
+        self.add_known_words()
+
+        if self.checked_file_path.endswith('.md'):
+            self.check_md_file()
+
+        elif self.checked_file_path.endswith('.yml'):
+            with open(self.checked_file_path, 'r') as yaml_file:
+                yml_info = yaml.safe_load(yaml_file)
+
+            self.check_yaml(yml_info=yml_info)
+
+        else:
+            print_error("\nThe file {} is not supported for spell-check command.\n"
+                        "Only .yml or .md files are supported.".format(self.checked_file_path))
+            return False
+
+        if len(self.unknown_words) > 0:
+            print_error(u"\nWords that might be misspelled were found in {}:\n\n{}".format(
+                self.checked_file_path, '\n'.join(self.unknown_words)))
+            return False
+
+        print_color("\nNo misspelled words found in {}".format(self.checked_file_path), LOG_COLORS.GREEN)
+        return True
+
+    def add_known_words(self):
+        # adding known words file if given - these words will not count as misspelled
+        if self.known_words:
+            known_words_file = open(self.known_words, 'r')
+            self.spellchecker.word_frequency.load_text_file(known_words_file)
```

Comment:
use with statement please
you are not closing the file and it will close it automatically when using this